filebeat/input/filestream/input_integration_test.go

```diff
@@ -45,17 +45,17 @@ func TestFilestreamCloseRenamed(t *testing.T) {
 	testlogName := "test.log"
 	inp := env.mustCreateInput(map[string]interface{}{
 		"paths":                                []string{env.abspath(testlogName) + "*"},
-		"prospector.scanner.check_interval":    "1ms",
+		"prospector.scanner.check_interval":    "10ms",
 		"close.on_state_change.check_interval": "1ms",
 		"close.on_state_change.renamed":        "true",
 	})
 
-	ctx, cancelInput := context.WithCancel(context.Background())
-	env.startInput(ctx, inp)
-
 	testlines := []byte("first log line\n")
 	env.mustWriteLinesToFile(testlogName, testlines)
 
+	ctx, cancelInput := context.WithCancel(context.Background())
```

Comment:
For my understanding, if the input is closed it triggers sending the pending events?

Reply:
No. If the input is closed the whole reading and publishing pipeline is cancelled. That is why first the test waits for all expected events to arrive and then the input is cancelled.